### ISSUE TITLE
bug : Adding PublicURL for HTTP redirection do not work as expected

### ISSUE THREAD

**Issue by mckaygerhard** [ViBiOh/fibr#152]
errr @ViBiOh  the commit  368580d  over the issue #146 improves the publicurl handler but only for full domain hijaking, not for subpath, by example:

1. running as : `fibr_linux_amd64  -csp ''  -noAuth -address "127.0.0.1"  -publicURL "http://domain/path1"  -port 10900    -fsDirectory /home/data/path1 -frameOptions "*"` cause a loop redirection cos the directory are same as the subpath

2.  running as: `fibr_linux_amd64  -csp ''  -noAuth -address "127.0.0.1"  -publicURL "http://domain/path1"  -port 10900    -fsDirectory /home/data/ -frameOptions "*"` do not ause loop redirection.. cos the subpath is a directory in the data dir, but any other links prefix the "path1" so no files working there

3.  running as: `fibr_linux_amd64   -noAuth -address "127.0.0.1"  -publicURL "http://sub.domain"  -port 10900    -fsDirectory /home/data/ ` do not do not work without `csp` and `frameOptions` wehen you use any cross domain path cos browser detect the reverse proxy redirection and is not the same domain if the reverse proxy are from main "domain" ("sub,.domain")

all this commit has nothing to do with issue #146

**Comment by ViBiOh** [comment]
I've added the `-pathPrefix` option. I've tested it extensively locally, let me know if it suits your need

e.g. 

```
fibr -pathPrefix "/fibr" -publicURL "https://mydomain.com" -noAuth
```

The command `curl http://localhost:1080/fibr/` will work ;)

**Comment by mckaygerhard** [comment]
I will test tomorrow, and as always post good feedback here!

**Comment by mckaygerhard** [comment]
umm tested, prefix works pretty well using the port .. but now i m pretty confused how to redirect my proxy lighttp webg server.. i need help

**Comment by mckaygerhard** [comment]
ok solved: for this we need fibr 0.4.1 release that fixed the prefuix paths, and a lighttpd or hiawata web server (powered lighty webservers), the idea is to handle security only in one port, and not let unnecesary monitoring of with fail2ban.. also we improve the performance handling all the service by lighty:

**1)** fibr must be invoked as:

`fibr -pathPrefix "/patfiles" -publicURL "https://domainfiles.com" -port 10900 .. `

so the resulting will be for files under "pepe" directory as `http://domainfiles.com:10900/patfiles/pepe`

**2)** for lighttp reverse proxy we must config as:

```
$HTTP["url"] =~ "(^/patfiles($|/))" {
   proxy.server  += ( "" => ("" => ( "host" => "0.0.0.0", "port" => 10900 ))) 
}
```

**3)** and now we can li8mit fibr by using of `-address "127.0.0.1"` in `fibr` to only handle all of the request by the lighttpd service, so fail2ban security can be handle only by the 80 port and not by unnecesary monitoring of 2 ports (fibr + lighttpd).

i hope this helps to improve security as also featured the performance.. later i will fill `/cods` with some "how tos" for alpine and debian so far.
